SU2_CFD/src/solvers/CEulerSolver.cpp

```diff
@@ -2431,20 +2422,22 @@ void CEulerSolver::Upwind_Residual(CGeometry *geometry, CSolver **solver_contain
           Project_Grad_j -= Vector_ij[iDim]*Gradient_j[iVar][iDim];
         }
 
-        if (limiter) {
-          if (van_albada) {
-            su2double V_ij = V_j[iVar] - V_i[iVar];
-            Limiter_i[iVar] = V_ij*( 2.0*Project_Grad_i + V_ij) / (4*pow(Project_Grad_i, 2) + pow(V_ij, 2) + EPS);
-            Limiter_j[iVar] = V_ij*(-2.0*Project_Grad_j + V_ij) / (4*pow(Project_Grad_j, 2) + pow(V_ij, 2) + EPS);
-          }
-          Primitive_i[iVar] = V_i[iVar] + Limiter_i[iVar]*Project_Grad_i;
-          Primitive_j[iVar] = V_j[iVar] + Limiter_j[iVar]*Project_Grad_j;
+        su2double lim_i = 1.0;
+        su2double lim_j = 1.0;
```

Comment:
Btw: The compiler (with optimization on) is clever enough to optimize out variable creation inside of these loops, right. So creating a static var inside of  an edge/point loop is fine? The unspoken rule (to some point in the past) was to put the var declaration outside of these loops. If there is no performance penalty to that than I also prefer the version you used here (I already do that better be the case).

Good job with the limiter: reads easier like that and all necessary code is close together

Reply:
I don't think the rule was unspoken, you had to do it for Fortran and at least early C.

Local variables are optimized away provided they do not involve dynamic allocation, or more formally that they don't have effects outside of local scope.
https://gcc.godbolt.org/z/Yx1hWT
Heap elision is tricky stuff (5 min video https://www.youtube.com/watch?v=FWD1msG8YU4&ab_channel=C%E1%90%A9%E1%90%A9WeeklyWithJasonTurner) and in general we cannot assume compilers will do it.

Reply:
Good to see you had a little `fun` :) and I should have thought about compiler-explorer myself to check that out. (Also means I have understand what happens in the assembler language to roughly understand what is happening.)
And interesting video on 'heap elision', I didn't know that existed.( And for the 10th time in a year I tell myself again: "Time to use clang for a bit." :) ).